### PR TITLE
[ADR]: React Component and Hooks Props Naming

### PR DESCRIPTION
**Pull request by yordis**

### CODE REVIEW

File: adrs/2131725034/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Understanding a React component API allows engineers to focus their attention on
+more important manners such as the business requirements.
+
+When it comes to debugging problems in production and minified code where
+finding where the problem is a challenge and time critical. Being able to reduce
+the cognitive load reading through the code base is crucial.
+
+Inconsistent properties naming promotes bikeshedding slowing down software
+development in code reviews and disagreements between developers.
```
Comment (TMiller00) [inline adrs/2131725034/README.md:31]: Might rewrite this as something like:

_Inconsistent property naming promotes bikeshedding, which slows down software development cycles in code review and results in unnecessary disagreements between developers._

also, TIL I learned bikeshedding

File: adrs/2131725034/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* Callbacks (Events, Control Flow)
+
+Data properties are primarily used for displaying information or transforming
+into another set of data.
+
+While Callbacks are used to control the flow of data such as filtering, sorting,
+for example: `onFilterAccount`, `onSortBy`, etc. Or reacting to events, the
+browser already tries to make distinction between these two categories by
+prefixing the events handlers with `on`, for example: `onClick`, `onChange`,
+etc.
```
Comment (TMiller00) [inline adrs/2131725034/README.md:21]: Might rewrite this as:

_Callbacks are used to control the flow of data. Functions such as `onFilterAccount` and `onSortBy` would be used to filter and sort data.  Callbacks can also react to events. The browser already makes a distinction between data and callback properties by prefixing the event handlers with `on`, for example: `onClick`, `onChange`, etc._

File: adrs/2131725034/README.md (hunk 1 above)
Comment (yordis) [reply adrs/2131725034/README.md]: Fixed it, hopefully.

File: adrs/2131725034/README.md (hunk 2 above)
Comment (yordis) [reply adrs/2131725034/README.md]: Fixed it, hopefully.

**Review by mboperator — APPROVED** [review]
Love using the "on" pattern

File: adrs/2131725034/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+the cognitive load reading through the code base is crucial.
+
+Inconsistent property naming promotes bikeshedding, which slows down software
+development cycles in code review and results in unnecessary disagreements
+between developers.
+
+## Resolution
+
+* React Component and Hook `data` properties **MUST** be a noun.
+* React Component and Hook `callback` properties **MUST** a verb prefix
```
Comment (dancingshell) [inline adrs/2131725034/README.md:39]: is there another example other than `on`? Maybe `get` or `set` or `update`?

```suggestion
* React Component and Hook `callback` properties **MUST** have a verb prefix
```

Comment (yordis) [reply adrs/2131725034/README.md]: It depends, normally, we have some called [CQS](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation) where the norm is to use `Get` for all the Query functions, which I would suggest depending on the context, I have been trying to stick to it for clarity about what my functions do.

But I am not sure about this context.

What do you have in mind?

**Review by tleef — APPROVED** [review]
Looks good to me

**Review by TMiller00 — APPROVED** [review]
Screen Shot 2019-04-19 at 11 14 29 AM

**Comment by manuphatak** [comment]
Love it!  I give it the post-merge +1 👍

**Pull request merged**